### ISSUE TITLE
Generator function causes crashing in the debug mode.

### ISSUE THREAD

**Issue by xiaxinmeng** [exaloop/codon#366]
The following code defines a generator function g that takes two arguments x and y, and yields tuples of values formed by iterating over each element of x and y. In the release mode of Code, this code can work normally while in a debug mode, codon reports a segfault.

test.py 
```
def g(x, y):
  for i in x:
    for j in y:
      yield i, j

r2 = (0, 1)
result = []
for e in g(r2, g(r2, r2)):
    result.append(e)
print(result)
```

Behavior on debug mode
\> codon run test.py
>Segmentation fault

Behavior on release mode
\> codon run --release test.py
> [(0, (0, 0)), (0, (0, 1)), (0, (1, 0)), (0, (1, 1))]

Environment:
Ubuntu 18.04
Codon v0.16.0
CPython 3.10.8

**Comment by elisbyberi** [comment]
@xiaxinmeng You are attempting to iterate over a generator that has already been exhausted. Duplicate of #246 

``` python
def g(x, g):  # "g: tuple[int, int]" or "g: Generator[Tuple[int,int]]"
    y = [i for i in g]  # exhaust generator only once
    for i in x:
        for j in y:
            yield i, j


x = (0, 1)
result = []
for e in g(x, g(x, x)):
    result.append(e)

print(result)
```
prints:
`[(0, (0, 0)), (0, (0, 1)), (0, (1, 0)), (0, (1, 1)), (1, (0, 0)), (1, (0, 1)), (1, (1, 0)), (1, (1, 1))]`

**Comment by inumanag** [comment]
Duplicate of #246 .
